yt/frontends/ramses/fields.py

```diff
@@ -241,35 +256,83 @@ def create_cooling_fields(self):
         filename = "%s/cooling_%05i.out" % (
             os.path.dirname(self.ds.parameter_filename), int(num))
 
-        if not os.path.exists(filename): return
-        def _create_field(name, interp_object):
+        if not os.path.exists(filename): 
+            print("This output has no cooling fields")
+            return
+
+        #Function to create the cooling fields
+        def _create_field(name, interp_object,unit):
             def _func(field, data):
                 shape = data["temperature"].shape
                 d = {'lognH': np.log10(_X*data["density"]/mh).ravel(),
                      'logT' : np.log10(data["temperature"]).ravel()}
-                rv = 10**interp_object(d).reshape(shape)
-                # Return array in unit 'per volume' consistently with line below
-                return data.ds.arr(rv, number_density_unit)
+                rv = interp_object(d).reshape(shape)
+                if not(name[-1]=='mu'):
```

Comment:
(Does not need to change!) Style nit: I think we usually prefer more along the lines of `if name[-1] != "mu"`.